Core/EventHandlers.lua

```diff
@@ -1728,6 +1728,21 @@ function R:OnEvent(event, ...)
 		Rarity.isFishing = false
 		Rarity.isPool = false
 
+		--Handle Disgusting vat fishing
+		if Rarity.relevantSpells[405274] == "Disgusting Vat Fishing" and (Rarity.lastNode == L["Disgusting Vat"]) then
```

Comment:
What is the purpose of this condition? The first part is always true (spell is part of the database), isn't it?

Looking at the other nodes, it seems `previousSpell` might be used to make sure the spell that was cast is the one we want (here, Disgusting Vat Fishing)... Not sure if that's sufficient in this case, though.

Reply:
iirc i tried previous spell first but couldnt get it too work, was probably forgetting to add something, not sure

Reply:
Just to recap: This condition is pointless and will be removed. The problem with the current/previous spell variables was that `UNIT_SPELLCAST_SENT` is required to update them, but fishing from the vat doesn't trigger this event. Presumably, it's a special case where Fishing (the spell) is initiated by the server after interacting with the object itself, and not by the client.